gobblin-data-management/src/test/java/org/apache/gobblin/data/management/copy/iceberg/IcebergRegisterStepTest.java

```diff
@@ -75,6 +71,66 @@ protected IcebergCatalog createDestinationCatalog() throws IOException {
     }
   }
 
+  @Test
+  public void testRegisterIcebergTableWithRetryer() throws IOException {
+    TableMetadata justPriorDestTableMetadata = mockTableMetadata("foo", "bar");
+    TableMetadata readTimeSrcTableMetadata = Mockito.mock(TableMetadata.class);
+    IcebergTable mockTable = mockIcebergTable("foo", "bar");
+    // Mocking registerIcebergTable() call to fail for first two attempts and then succeed
+    // So total number of invocations to registerIcebergTable() should be 3 only
+    Mockito.doThrow(new RuntimeException())
+        .doThrow(new RuntimeException())
+        .doNothing()
+        .when(mockTable).registerIcebergTable(any(), any());
+    IcebergRegisterStep regStep = createIcebergRegisterStepInstance(readTimeSrcTableMetadata, justPriorDestTableMetadata, mockTable, new Properties());
+    try {
+      regStep.execute();
+      Mockito.verify(mockTable, Mockito.times(3)).registerIcebergTable(any(), any());
+    } catch (RuntimeException re) {
+      Assert.fail("Got Unexpected Runtime Exception", re);
+    }
+  }
+
+  @Test
+  public void testRegisterIcebergTableWithDefaultRetryerConfig() throws IOException {
+    TableMetadata justPriorDestTableMetadata = mockTableMetadata("foo", "bar");
+    TableMetadata readTimeSrcTableMetadata = Mockito.mock(TableMetadata.class);
+    IcebergTable mockTable = mockIcebergTable("foo", "bar");
+    // Mocking registerIcebergTable() call to always throw exception
+    Mockito.doThrow(new RuntimeException()).when(mockTable).registerIcebergTable(any(), any());
+    IcebergRegisterStep regStep = createIcebergRegisterStepInstance(readTimeSrcTableMetadata, justPriorDestTableMetadata, mockTable, new Properties());
+    try {
+      regStep.execute();
+      Assert.fail("Expected Runtime Exception");
+    } catch (RuntimeException re) {
+      // The default number of retries is 5 so register iceberg table should fail after retrying for 5 times
+      String msg = String.format("Failed to register iceberg table : (src: {%s}) - (dest: {%s}) : (retried %d times)", srcTableId, destTableId, 5);
+      Assert.assertTrue(re.getMessage().startsWith(msg), re.getMessage());
+    }
+  }
```

Comment:
bonus points if you can refactor this into a helper method that takes config props (possibly empty) plus num retries expected, then performs the validation.  keep two tests - default vs. override of num retries - but have a common validation impl that doesn't repeat itself

Reply:
Added method for validating retry log statement but done without passing any config.
In future we can add props and list of values, if multiple log statements needs to be validated.